textures/src/main/java/com/dumbdogdiner/stickyapi/util/textures/TextureHelper.java

```diff
@@ -0,0 +1,259 @@
+     */
+    public static Map<String, String> getTextureMapCategory(@NotNull String cat) {
+        return TextureMap.get(cat.toUpperCase());
+    }
+
+
+    /**
+     * Get a list of all the textures for a given Category
+     *
+     * @param cat a Category of textures
+     * @return a {@link List} of the textures in a given Category
+     * @see #getCategories()
+     */
+    public static @NotNull List<String> getTextures(@NotNull String cat) {
+        return Collections.unmodifiableList(new ArrayList<>(getTextureMapCategory(cat).keySet()));
+    }
+
+    /**
+     * Gets a texture string given a category and texture name
+     *
+     * @param cat  The texture category
+     * @param name The texture name
+     * @return The texture string matching
+     * @throws NoSuchElementException if the specified texture is not found
+     */
+    public static String getTexture(@NotNull String cat, @NotNull String name) throws NoSuchElementException {
+        return TextureMap.get(cat.toUpperCase()).get(name.toUpperCase());
+    }
+
+    /**
+     * Gets a texture string given a qualified name
+     *
+     * @param qualifiedName A given texture as a qualified name
+     * @return the texture string for the given texture
+     * @throws NoSuchElementException if the specified texture is not found
+     * @see #toQualifiedName(String, String)
+     */
+    public static @Nullable String getTexture(@NotNull String qualifiedName) throws NoSuchElementException {
+        String[] splits = qualifiedName.split("\\.");
+        if (splits.length != 2 && splits.length != 1)
+            throw new RuntimeException("Invalid qualified name: " + qualifiedName);
+        if (splits[0].equals("*")) {
+            @Nullable String texture = null;
+            for (@NotNull String cat : getCategories()) {
+                try {
+                    texture = getTexture(cat, qualifiedName);
+                } catch (NoSuchElementException e) {
+                    continue;
+                }
+                return texture;
+            }
+            return texture;
+        } else {
+            return getTexture(splits[0], splits[1]);
+        }
+    }
+
+    /**
+     * Returns a {@link List} of Qualified Names, of the format of "{CATEGORY}.{TEXTURE}"
+     *
+     * @return a {@link List} of Qualified Names of all textures
+     */
+    public static @NotNull List<String> getQualifiedNames() {
+        @NotNull ArrayList<String> qualifiedNames = new ArrayList<>();
+        for (@NotNull String cat : getCategories()) {
+            for (String texture : getTextureMapCategory(cat).keySet()) {
+                qualifiedNames.add(toQualifiedName(cat, texture));
+            }
+        }
+        return Collections.unmodifiableList(qualifiedNames);
+    }
+
+    /**
+     * @see TextureHelper#toTextureJson(URL)
+     */
+    public static @NotNull JsonObject toTextureJson(@NotNull URL url) {
+        return toTextureJson(url.toExternalForm());
```

Comment:
For clarity's sake
```suggestion
        return this.toTextureJson(url.toExternalForm());
```

Reply:
I find it clearer without the this for methods

Reply:
I suspect a lot of us would disagree with you there. Please use `this` where you can :3